src/providers/AppProvider.tsx

```diff
@@ -3,15 +3,15 @@
 import { AuthKitProvider } from "@farcaster/auth-kit"
 import "@farcaster/auth-kit/styles.css"
 
-if (!process.env.NEXT_PUBLIC_APP_DOMAIN) {
-  throw new Error("Please define NEXT_PUBLIC_APP_DOMAIN in env.")
+if (!process.env.NEXT_PUBLIC_VERCEL_URL) {
```

Comment:
`NEXT_PUBLIC_VERCEL_URL` is a default system URL inside vercel so it can help us define it per preview link!